Coding Standards

There may be some coding standards that can be fixed. With GHA running the latest result was:

## PHPStan

```
Error: Method XMLSitemapTestHelper::setUp() should return bool but return statement is missing.
Error: Method XMLSitemapFunctionalTest::setUp() should return bool but return statement is missing.
Error: Method XMLSitemapRobotsTxtIntegrationTest::setUp() should return bool but return statement is missing.
Error: Method XmlsitemapViewsFieldsTest::setUp() should return bool but return statement is missing.
Error: Function drush_print not found.
Error: Function dt not found.
Error: Function drush_get_option not found.
Error: Function drush_set_error not found.
Error: Function dt not found.
Error: Function drush_log not found.
Error: Function dt not found.
Error: Function drush_print not found.
Error: Function dt not found.
Error: Function drush_get_option not found.
Error: Function drush_print not found.
Error: Function dt not found.
Error: Function drush_print not found.
Error: Function dt not found.
Error: Function drush_get_option not found.
Error: Function drush_set_error not found.
Error: Function dt not found.
Error: Function drush_log not found.
Error: Function dt not found.
Error: Function drush_get_option not found.
Error: Function drush_log not found.
Error: Function dt not found.
Error: Function drush_log not found.
Error: Function dt not found.
Error: Function redirect_fetch_rids_by_path not found.
Error: Function redirect_fetch_rids_by_path not found.
Error: Method XMLSitemapWriter::startElement() should return bool but return statement is missing.
Error: Method XMLSitemapWriter::writeElement() should return bool but return statement is missing.
Error: Method XMLSitemapCustomFunctionalTest::setUp() should return bool but return statement is missing.
Error: Function drush_print not found.
Error: Function dt not found.
Error: Function drush_print not found.
Error: Function dt not found.
Error: Method XMLSitemapEnginesFunctionalTest::setUp() should return bool but return statement is missing.
Error: Method XMLSitemapI18nWebTestCase::setUp() should return bool but return statement is missing.
Error: Method XMLSitemapI18nNodeTest::setUp() should return bool but return statement is missing.
Error: Method XMLSitemapMenuFunctionalTest::setUp() should return bool but return statement is missing.
Error: Method XMLSitemapNodeFunctionalTest::setUp() should return bool but return statement is missing.
Error: Method XMLSitemapTaxonomyFunctionalTest::setUp() should return bool but return statement is missing.
Error: Method XMLSitemapUserFunctionalTest::setUp() should return bool but return statement is missing.
Error: Process completed with exit code 1.
```

## PHPCS

```
Error: No space found before comment text; expected "// $this->_assertNoWatchdogErrors();" but found "//$this->_assertNoWatchdogErrors();"
Error: The array declaration extends to column 168 (the limit is 80). The array content should be split up over multiple lines
Error: Return type missing for @return tag in function comment
Error: The array declaration extends to column 158 (the limit is 80). The array content should be split up over multiple lines
Error: Line indented incorrectly; expected at least 4 spaces, found 0
Error: The array declaration extends to column 116 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 116 (the limit is 80). The array content should be split up over multiple lines
Error: Missing function doc comment
Error: Visibility must be declared on method "testAssertFlag"
Error: The array declaration extends to column 120 (the limit is 80). The array content should be split up over multiple lines
Error: Inline doc block comments are not allowed; use "/* Comment */" or "// Comment" instead
Error: Line exceeds maximum limit of 120 characters; contains 166 characters
Error: There must be no blank lines before the file comment
Error: Missing class doc comment
Error: The array declaration extends to column 148 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 154 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 164 (the limit is 80). The array content should be split up over multiple lines
Error: There must be no blank lines before the file comment
Error: The array declaration extends to column 116 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 88 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 94 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 117 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 113 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 91 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 114 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 299 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 89 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 222 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 229 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 167 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 197 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 119 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 82 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 99 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 229 (the limit is 80). The array content should be split up over multiple lines
Error: There must be no blank lines before the file comment
Error: There must be no blank lines before the file comment
Error: The array declaration extends to column 98 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 93 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 81 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 91 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 91 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 83 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 159 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 165 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 116 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 106 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 144 (the limit is 80). The array content should be split up over multiple lines
Error: Missing function doc comment
Error: The array declaration extends to column 86 (the limit is 80). The array content should be split up over multiple lines
Error: The array declaration extends to column 175 (the limit is 80). The array content should be split up over multiple lines
Error: Return comment indentation must be 3 spaces, found 1 spaces
Error: Return comment indentation must be 3 spaces, found 1 spaces
Error: Process completed with exit code 2.
```

--- a/.github/phpstan/baseline.neon
+++ b/.github/phpstan/baseline.neon
@@ -1,6 +1,11 @@
 parameters:
 	ignoreErrors:
 		-
-			message: "#^$#"
+			message: "#^Method XMLSitemapWriter\\:\\:startElement\\(\\) should return bool but return statement is missing\\.$#"
 			count: 1
-			path: ../../xmlsitemap.module
+			path: ../../xmlsitemap.xmlsitemap.inc
+
+		-
+			message: "#^Method XMLSitemapWriter\\:\\:writeElement\\(\\) should return bool but return statement is missing\\.$#"
+			count: 1
+			path: ../../xmlsitemap.xmlsitemap.inc

--- a/.github/phpstan/phpstan.neon
+++ b/.github/phpstan/phpstan.neon
@@ -2,8 +2,8 @@
 # don't need to show up in every report.
 # If a problem got fixed, the file should be re-generated.
 # @see https://phpstan.org/user-guide/baseline
-# includes:
-#  - baseline.neon
+includes:
+  - baseline.neon
 # @see https://phpstan.org/config-reference
 parameters:
   level: 0
@@ -18,3 +18,4 @@ parameters:
   excludePaths:
     - *.tpl.php
     - *.api.php
+    - *.drush.inc

--- a/tests/xmlsitemap.test
+++ b/tests/xmlsitemap.test
@@ -29,6 +29,7 @@ class XMLSitemapTestHelper extends BackdropWebTestCase {
   public function setUp($modules = array()) {
     array_unshift($modules, 'xmlsitemap');
     parent::setUp($modules);
+    return TRUE;
   }
 
   /**
@@ -860,6 +861,7 @@ class XMLSitemapFunctionalTest extends XMLSitemapTestHelper {
       'administer xmlsitemap',
     ));
     $this->backdropLogin($this->admin_user);
+    return TRUE;
   }
 
   /**
@@ -946,6 +948,7 @@ class XMLSitemapRobotsTxtIntegrationTest extends XMLSitemapTestHelper {
   public function setUp($modules = array()) {
     $modules[] = 'robotstxt';
     parent::setUp($modules);
+    return TRUE;
   }
 
   /**

--- a/tests/xmlsitemap_views_fields.test
+++ b/tests/xmlsitemap_views_fields.test
@@ -31,6 +31,7 @@ class XmlsitemapViewsFieldsTest extends BackdropWebTestCase {
     $modules[] = 'views';
     $modules[] = 'xmlsitemap_node';
     parent::setUp($modules);
+    return TRUE;
   }
 
   /**

--- a/xmlsitemap.generate.inc
+++ b/xmlsitemap.generate.inc
@@ -205,8 +205,8 @@ function xmlsitemap_generate_chunk(stdClass $sitemap, XMLSitemapWriter $writer, 
     $parsed_url = backdrop_parse_url($link['loc']);
     // Skip nodes which are 301 redirected.
     if (config_get('xmlsitemap.settings', 'redirect')) {
-      $relative_redirect = redirect_fetch_rids_by_path($link['loc'], $link['language']->language, TRUE);
-      $alias_redirect = redirect_fetch_rids_by_path(ltrim(url($link['loc']), '/'), $link['language']->language, TRUE);
+      $relative_redirect = redirect_load_by_source($link['loc'], $link['language']->langcode, TRUE);
+      $alias_redirect = redirect_load_by_source(ltrim(url($link['loc']), '/'), $link['language']->langcode, TRUE);
       // If node contains a 301 redirect we skip it.
       if (!empty($relative_redirect) || !empty($alias_redirect)) {
         continue;

--- a/xmlsitemap.generate.inc
+++ b/xmlsitemap.generate.inc
@@ -205,8 +205,8 @@ function xmlsitemap_generate_chunk(stdClass $sitemap, XMLSitemapWriter $writer, 
     $parsed_url = backdrop_parse_url($link['loc']);
     // Skip nodes which are 301 redirected.
     if (config_get('xmlsitemap.settings', 'redirect')) {
-      $relative_redirect = redirect_load_by_source($link['loc'], $link['language']->langcode, TRUE);
-      $alias_redirect = redirect_load_by_source(ltrim(url($link['loc']), '/'), $link['language']->langcode, TRUE);
+      $relative_redirect = redirect_load_by_source($link['loc'], $link['language']->langcode, array());
+      $alias_redirect = redirect_load_by_source(ltrim(url($link['loc']), '/'), $link['language']->langcode, array());
       // If node contains a 301 redirect we skip it.
       if (!empty($relative_redirect) || !empty($alias_redirect)) {
         continue;

--- a/xmlsitemap_custom/tests/xmlsitemap_custom.test
+++ b/xmlsitemap_custom/tests/xmlsitemap_custom.test
@@ -22,6 +22,7 @@ class XMLSitemapCustomFunctionalTest extends XMLSitemapTestHelper {
 
     $this->admin_user = $this->backdropCreateUser(array('access content', 'administer xmlsitemap'));
     $this->backdropLogin($this->admin_user);
+    return TRUE;
   }
 
   /**

--- a/xmlsitemap_engines/tests/xmlsitemap_engines.test
+++ b/xmlsitemap_engines/tests/xmlsitemap_engines.test
@@ -30,6 +30,7 @@ class XMLSitemapEnginesFunctionalTest extends XMLSitemapTestHelper {
     $this->backdropLogin($this->admin_user);
 
     $this->submit_url = url('ping', array('absolute' => TRUE, 'query' => array('sitemap' => ''))) . '[sitemap]';
+    return TRUE;
   }
 
   /**

--- a/xmlsitemap_i18n/tests/xmlsitemap_i18n.test
+++ b/xmlsitemap_i18n/tests/xmlsitemap_i18n.test
@@ -54,6 +54,7 @@ class XMLSitemapI18nWebTestCase extends XMLSitemapTestHelper {
     $sitemap = new stdClass();
     $sitemap->context = array('language' => 'fr');
     xmlsitemap_sitemap_save($sitemap);
+    return TRUE;
   }
 
 }
@@ -145,6 +146,7 @@ class XMLSitemapI18nNodeTest extends XMLSitemapI18nWebTestCase {
     config_set('i18n.settings', 'language_content_type_page', 1);
     $this->admin_user = $this->backdropCreateUser(array('administer nodes'));
     $this->backdropLogin($this->admin_user);
+    return TRUE;
   }
 
   /**

--- a/xmlsitemap_menu/tests/xmlsitemap_menu.test
+++ b/xmlsitemap_menu/tests/xmlsitemap_menu.test
@@ -38,6 +38,7 @@ class XMLSitemapMenuFunctionalTest extends XMLSitemapTestHelper {
 
     $this->admin_user = $this->backdropCreateUser(array('administer menu', 'administer xmlsitemap'));
     $this->normal_user = $this->backdropCreateUser(array('access content'));
+    return TRUE;
   }
 
   /**

--- a/xmlsitemap_node/tests/xmlsitemap_node.test
+++ b/xmlsitemap_node/tests/xmlsitemap_node.test
@@ -52,6 +52,7 @@ class XMLSitemapNodeFunctionalTest extends XMLSitemapTestHelper {
       'view own unpublished content',
     ));
     xmlsitemap_link_bundle_settings_save('node', 'page', array('status' => 1, 'priority' => 0.5));
+    return TRUE;
   }
 
   /**

--- a/xmlsitemap_taxonomy/tests/xmlsitemap_taxonomy.test
+++ b/xmlsitemap_taxonomy/tests/xmlsitemap_taxonomy.test
@@ -38,6 +38,7 @@ class XMLSitemapTaxonomyFunctionalTest extends XMLSitemapTestHelper {
 
     $this->admin_user = $this->backdropCreateUser(array('administer taxonomy', 'administer xmlsitemap'));
     $this->normal_user = $this->backdropCreateUser(array('access content'));
+    return TRUE;
   }
 
   /**

--- a/xmlsitemap_user/tests/xmlsitemap_user.test
+++ b/xmlsitemap_user/tests/xmlsitemap_user.test
@@ -49,6 +49,7 @@ class XMLSitemapUserFunctionalTest extends XMLSitemapTestHelper {
     // Update the normal user to make its sitemap link visible.
     $account = clone $this->normal_user;
     user_save($account, array('access' => 1, 'login' => 1));
+    return TRUE;
   }
 
   /**

